api/ask_astro/slack/controllers/feedback/bad.py

```diff
@@ -10,54 +12,81 @@
+
+    :param body: The slack event body.
+    :param ack: Acknowledgement object from slack_bolt.
+    :param respond: Response object from slack_bolt.
+    :param client: Slack API client.
+    """
+    await ack()
+
+    details = extract_feedback_details(body)
+    if not details:
+        return
+
+    request_id = details["value"].split(":")[0]
+    await submit_feedback(request_id, False, source_info={"type": "slack", "user": details["user"]})
 
     async with TaskGroup() as tg:
-        tg.create_task(
-            respond(
-                f"😥 Thank you for your feedback, <@{user}>!",
-                thread_ts=thread_ts,
-                replace_original=False,
-                response_type="ephemeral",
-            )
-        )
-
-        async def update_reaction():
-            try:
-                await client.reactions_add(
-                    name="thumbsdown",
-                    channel=channel,
-                    timestamp=message_ts,
-                )
-            except SlackApiError as e:
-                # ignore the error if the reaction already exists
-                if e.response["error"] != "already_reacted":
-                    raise e
-
-            try:
-                await client.reactions_remove(
-                    name="thumbsup",
-                    channel=channel,
-                    timestamp=message_ts,
-                )
-            except Exception:
-                pass
-
-        tg.create_task(
-            update_reaction(),
-        )
+        tg.create_task(_send_response(details, respond))
+        tg.create_task(_update_reaction(details, client))
+
+
+async def _send_response(details: dict[str, str], respond: AsyncRespond) -> None:
+    """
+    Send a response back to the user in Slack.
+
+    :param details: The details extracted from the Slack body.
+    :param respond: Response object from slack_bolt.
+    """
+    await respond(
+        "😥 Thank you for your feedback, <@%s>!" % details["user"],
+        thread_ts=details["thread_ts"],
+        replace_original=False,
+        response_type="ephemeral",
+    )
+
+
+async def _update_reaction(details: dict[str, str], client: AsyncWebClient) -> None:
+    """
+    Add a 'thumbsdown' reaction and remove the 'thumbsup' reaction from the original message.
+
+    :param details: The details extracted from the Slack body.
+    :param client: Slack API client.
+    """
+    try:
+        await client.reactions_add(name="thumbsdown", channel=details["channel"], timestamp=details["message_ts"])
+    except SlackApiError as e:
+        # ignore the error if the reaction already exists
+        if e.response["error"] != "already_reacted":
+            raise e
+
+    try:
+        await client.reactions_remove(name="thumbsup", channel=details["channel"], timestamp=details["message_ts"])
+    except Exception as e:
+        logger.debug("Failed to remove thumbsup reaction: %s", e)
+        pass
```

Comment:
```suggestion
```

Not sure whether we need this `pass`

Reply:
We pass it as there is already a thumbs up

Reply:
removed it though